### ISSUE TITLE
linux-tkg-pds 5.8.9 breaks mdadm RAID detection

### ISSUE THREAD

**Issue by z0n** [Frogging-Family/linux-tkg#70]
Hi, I've been using the kernel for quite a while without issues. Yesterday, I updated from linux-tkg-pds 5.8.7 to 5.8.9.
After rebooting, my RAID is not detected by mdadm anymore. `mdadm --assemble --scan` doesn't find anything.  
I then booted the default Arch Linux 5.8.9 kernel and my RAID is detected fine there (see sdc and sdd):
```
NAME        MAJ:MIN RM   SIZE RO TYPE  MOUNTPOINT
sda           8:0    0 465.8G  0 disk  
├─sda1        8:1    0   300M  0 part  
├─sda2        8:2    0    99M  0 part  
├─sda3        8:3    0   128M  0 part  
├─sda4        8:4    0 464.6G  0 part  
└─sda5        8:5    0   616M  0 part  
sdb           8:16   0   1.8T  0 disk  
├─sdb1        8:17   0    16M  0 part  
├─sdb2        8:18   0 488.3G  0 part  
└─sdb3        8:19   0   1.3T  0 part  /mnt/data_ssd
sdc           8:32   0   3.6T  0 disk  
└─md126       9:126  0   3.6T  0 raid1 
  ├─md126p1 259:4    0   128M  0 part  
  └─md126p2 259:5    0   3.6T  0 part  
sdd           8:48   0   3.6T  0 disk  
└─md126       9:126  0   3.6T  0 raid1 
  ├─md126p1 259:4    0   128M  0 part  
  └─md126p2 259:5    0   3.6T  0 part  
nvme0n1     259:0    0 476.9G  0 disk  
├─nvme0n1p1 259:1    0   512M  0 part  /boot
├─nvme0n1p2 259:2    0    16G  0 part  [SWAP]
└─nvme0n1p3 259:3    0 460.4G  0 part  /
```
Compare this to linux-tkg-pds:
```
sdc           8:32   0   3.6T  0 disk 
├─sdc1        8:33   0   128M  0 part 
└─sdc2        8:34   0   3.6T  0 part 
sdd           8:48   0   3.6T  0 disk 
├─sdd1        8:49   0   128M  0 part 
└─sdd2        8:50   0   3.6T  0 part
```

My setup:
OS: Arch Linux
CPU: Intel Core i7-8700K
Mainboard: Gigabyte Z370 AORUS Gaming 7
RAM: 32 GB DDR4-3000
GPU: NVIDIA GeForce GTX 1080 Ti

[Here's](https://pastebin.com/tkdkxDXE) my linux-tkg config. My /etc/mdadm.conf only contains `DEVICE partitions` which scans all partitions at boot.

**Comment by AdelKS** [comment]
What would be interesting for you to try, is to find the kernel's `.config` file that Arch uses for its default 5.8.9 version. And use it to replace the one in `linux58-tkg-config/config.x86_64`, because the script uses that one: it moves it inside the kernel's root tree and renames it to `.config`.

So here's what I suggest you to do: start your computer with the vanilla arch kernel, the config file is at `/proc/config.gz`. Use it to replace `linux58-tkg-config/config.x86_64`, then do the usual `makepkg -si`. If this solves your problem, then we can do something about it.

For information: the `.config` file is the file that is/should be present in at the root of the kernel's sources so it can compile, which is generated/updated with `make menuconfig` or `make xconfig`

**Comment by z0n** [comment]
@AdelKS No need to fix anything, I just updated to linux-tkg-pds 5.8.10 and it's working again :slightly_smiling_face:

**Comment by AdelKS** [comment]
Great!

**Comment by z0n** [comment]
@AdelKS Out of curiosity, I tried your suggestion (with 5.8.9) and it did not solve my issue. Not what then could've caused the issue (one of the patches I guess?) but I would invest more time in it as it's fixed with 5.8.10 anyway.

**Comment by Tk-Glitch** [comment]
-tkg kernel comes with vanilla Archlinux config, so it was likely an upstream issue (possibly with a cherrypicked fix on the Arch side if 5.8.9-arch* was fine).

**Comment by Tk-Glitch** [comment]
Yup, found it : https://git.archlinux.org/linux.git/commit/?h=v5.8.9-arch2&id=90b434f4088c347959313d01d46e207a00949f35
This was cherrypicked due to an upstream regression. All clear.
